core/src/main/java/org/apache/calcite/sql2rel/SqlToRelConverter.java

```diff
@@ -683,6 +683,13 @@ protected void convertSelectImpl(
     convertFrom(
         bb,
         select.getFrom());
+    if (RelOptUtil.isOrder(bb.root)
```

Comment:
this complex 'if' needs a comment explaining what it is doing.

Remove the space before ')'.

Reply:
I rewritten and added some comments to explain the code.